### PR TITLE
Cleanup request/response recycle #4711

### PR DESCRIPTION
**Pull request by gregw**
Reordered recycle in request and response to field order so that we can check that all fields are recycled for #4711

### CODE REVIEW

**Comment by gregw** [comment]
merge to 10 is here: https://github.com/eclipse/jetty.project/tree/jetty-10.0.x-4711-recylce-cleanup

File: jetty-server/src/main/java/org/eclipse/jetty/server/Request.java [hunk 1]
```diff
@@ -1857,37 +1857,40 @@ public boolean hasMetaData()
...
             throw new IllegalStateException("Request in context!");
-
-        if (_inputState == INPUT_READER)
+        if (_reader != null && _inputState == INPUT_READER)
         {
             try
             {
                 int r = _reader.read();
                 while (r != -1)
-                {
```
Comment (sbordet) [inline jetty-server/src/main/java/org/eclipse/jetty/server/Request.java]: `while` loops require braces for checkstyle.

File: jetty-server/src/main/java/org/eclipse/jetty/server/Request.java [hunk 2]
```diff
@@ -1857,37 +1857,40 @@ public boolean hasMetaData()
...
 
-        _dispatcherType = null;
-        setAuthentication(Authentication.NOT_CHECKED);
         getHttpChannelState().recycle();
-        if (_async != null)
-            _async.reset();
-        _async = null;
+        _requestAttributeListeners.clear();
+        _input.recycle();
+
```
Comment (sbordet) [inline jetty-server/src/main/java/org/eclipse/jetty/server/Request.java]: Unnecessary newline.

**Review by sbordet — COMMENTED** [review]
LGTM apart formatting changes.

**Review by sbordet — APPROVED** [review]

**Pull request merged**